src/modules/recipes/NewRecipeForm/index.tsx

```diff
@@ -0,0 +1,240 @@
+              <option key={area} value={area}>
+                {area}
+              </option>
+            ))}
+          </select>
+        </div>
+
+        <div className="form-group">
+          <label>Ingredients</label>
+          {fields.map((field, index) => (
+            <div key={field.id} className="ingredient-item">
+              <input
+                {...register(`ingredients.${index}.ingredient`, {
+                  required: "Ingredient is required.",
+                })}
+                placeholder="Ingredient"
+              />
+              <input
+                type="text"
+                {...register(`ingredients.${index}.amount`, {
+                  required: "Amount is required.",
+                  pattern: {
+                    value: /^(\d+(\.\d+)?|\d+(\,\d+)?)/, // Hem nokta hem virgül ile ondalıklı sayılara izin verir.
+                    message: "Please enter a valid amount.",
+                  },
+                })}
+                placeholder="Amount"
+              />
+              <select {...register(`ingredients.${index}.measure`)}>
+                <option value="">Measure</option>
+                <option value="grams">Grams</option>
+                <option value="liters">Liters</option>
+                <option value="pieces">Pieces</option>
+              </select>
+              <button
+                type="button"
+                className="remove-button"
+                onClick={() => remove(index)}
+              >
+                Remove
+              </button>
+            </div>
+          ))}
+          <button
+            type="button"
+            className="add-button"
+            onClick={() => append({ ingredient: "", measure: "", amount: "" })}
+          >
+            Add Ingredient
+          </button>
+        </div>
+
+        <div className="form-group">
+          <label htmlFor="strInstructions">Instructions</label>
+          <textarea
+            id="strInstructions"
+            {...register("strInstructions", {
+              required: "Instructions are required.",
+            })}
+          />
+          {errors.strInstructions && (
+            <p className="error-message">{errors.strInstructions.message}</p>
+          )}
+        </div>
+
+        <div className="form-group">
+          <label htmlFor="strMealImage">Meal Image URL</label>
+          <input
+            id="strMealImage"
+            {...register("strMealImage", {
+              pattern: {
+                value: /https?:\/\/.+\.(jpg|jpeg|png|webp|gif|bmp|tiff|svg)/,
+                message: "Please enter a valid URL for an image.",
+              },
+            })}
+          />
+          {errors.strMealImage && (
+            <p className="error-message">{errors.strMealImage.message}</p>
+          )}
+        </div>
+
+        <button type="submit" className="submit-button">
+          Submit
+        </button>
+      </form>
+    </div>
+  );
+};
+
+const categories = [
```

Comment:
It would be good to create TS enum values for categories and areas. E.g.:
```
export enum Category = {"Beef",
  "Chicken",
  "Dessert",
  "Lamb",
  "Miscellaneous",
  "Pasta",
  "Pork",
  "Seafood",
  "Side",
  "Starter",
  "Vegan",
  "Vegetarian",
  "Breakfast",
  "Goat",
  "Other"};

export const Categories: Category[] = Object.values(Category);

```

It should also be in the shared models file